### PR TITLE
Pawn

### PR DESCRIPTION
**Pull request by wdehaes**
Created the valid_move? method for pawn + rspec tests. Also made some corrections to obstructed? method in piece.rb (will change in rook PR as well). Based on PR:
- [ ] #36

### CODE REVIEW

File: app/models/pieces/pawn.rb [hunk 1]
```diff
@@ -1,2 +1,22 @@
 class Pawn < Piece
+  # rubocop:disable all
+  def valid_move?(x_destination, y_destination)
+    # check for Piece valid_move? conditions
+    return false unless super(x_destination, y_destination)
+    # a pawn can never move horizontally or, and never move more than two fields ahead, or more than one sideways
+    return false if moving_horizontally?(y_destination) || diff(x_destination, x_position) > 1 || diff(y_destination, y_position) > 2
```
Comment (lasley) [inline app/models/pieces/pawn.rb]: You can probably eliminate the `diff(y_destination, y_position) > 2` in favor of the explicit check later (except using `>=` instead).

File: app/models/pieces/pawn.rb [hunk 2]
```diff
@@ -1,2 +1,22 @@
...
+    #a pawn cannot move backwards
+    return false if (player.color == 'white' ? y_destination <= y_position : y_destination >= y_position)
+    # a pawn can only move 2 squares forward from its starting position
+    return false if diff(y_destination, y_position) == 2 && y_position != (player.color == 'white' ? 1 : 6)
+    if diff(x_destination, x_position).zero?
```
Comment (lasley) [inline app/models/pieces/pawn.rb]: The x and y diffs are being calculated a lot. Might be best to create an x_diff, y_diff intermediary var

File: app/models/pieces/pawn.rb [hunk 3]
```diff
@@ -1,2 +1,22 @@
 class Pawn < Piece
+  # rubocop:disable all
```
Comment (lasley) [inline app/models/pieces/pawn.rb]: It is best practice to only disable the specific cop that is failing & unable to be resolved. I assume it is ABCSize & PerceivedComplexity?

**Review by lasley — CHANGES_REQUESTED** [review]
comments inline

File: app/models/pieces/pawn.rb (hunk 1 above)
Comment (wdehaes) [reply app/models/pieces/pawn.rb]: I don't understand how that would work. My logic was that a move with diff_y > 2 is always invalid, but a move with diff_y == 2 can be valid in some cases.

File: app/models/pieces/pawn.rb (hunk 3 above)
Comment (wdehaes) [reply app/models/pieces/pawn.rb]: Yep

**Comment by erniejamison** [comment]
LGTM 👍

**Review by lasley — APPROVED** [review]

**Pull request merged**